tests/run-make/hir-tree/rmake.rs

```diff
@@ -0,0 +1,17 @@
+extern crate run_make_support;
+
+use run_make_support::{out_dir, rustc};
+
+// Test that hir-tree output doesn't crash and includes
+// the string constant we would expect to see.
+
+fn main() {
+    rustc()
+        .arg("-o")
+        .arg(&out_dir().join("input.hir"))
+        .arg("-Zunpretty=hir-tree")
+        .arg("input.rs")
+        .run();
+    let file = std::fs::read_to_string(&out_dir().join("input.hir")).unwrap();
```

Comment:
It feels like we should add helper functions for things like "assert file contains" etc., otherwise code like this will be repeated in every other rmake test :)

Reply:
> It feels like we should add helper functions for things like "assert file contains" etc., otherwise code like this will be repeated in every other rmake test :)

Yep, the intention is to build up a more useful / contributor-friendly support library as more run-make tests are ported over.

Reply:
My 2 cents: from my experience with writing similar "test DSLs", I would suggest to create these helpers as soon as possible in the process (basically the first time someone needs them, not only after N similar usages), and basically treat any imperative code in these high-level tests as a maintenance burden, basically making them as declarative as possible. Otherwise it might be a bit annoying to keep refactoring the tests later once such helpers are available. 

I would even discourage the `arg` method and create methods for setting things like `unpretty`, the output file etc., but this might be more controversial, as it will make the invoked command-line arguments more opaque.

Reply:
> My 2 cents: from my experience with writing similar "test DSLs", I would suggest to create these helpers as soon as possible in the process (basically the first time someone needs them, not only after N similar usages), and basically treat any imperative code in these high-level tests as a maintenance burden, basically making them as declarative as possible. Otherwise it might be a bit annoying to keep refactoring the tests later once such helpers are available.

Yeah, that makes a lot of sense.

> I would even discourage the arg method and create explicit methods for setting things like unpretty, the output file etc., but this might be more controversial, as it will make the invoked command-line arguments more opaque.

This is probably a good idea, if only just for documentation purposes.

Reply:
> This is probably a good idea, if only just for documentation purposes.

It would make it easier to modify all usages of these flags when the compiler changes its interface, and also easy to find all usages of these flags/parameters.

Also auto-blessing would be nice to have, although I'm not sure if it's actually very useful for run-make tests (they are useful if we check strict text content that changes fro time to time).

Reply:
I'm imagining something like:

```rs
rustc()
  .input_file(Path::new("input.rs"))
  .output_file(&out_dir().join("input.hir"))
  .unstable_option(UnstableOpt::Pretty { kind: PrettyKind::HirTree })
  .run()
```

Reply:
Yeah, that looks better. `input_file` can take `AsRef<Path>` to avoid the verbose `Path` constructor, and I would also maybe just add direct shortcuts for commonly occurring options, e.g. `.unpretty(PrettyKind::HirTree)` (I'm still not sure if this is a better idea than `-Zunpretty=hir`, since the latter is much better greppable, across all kinds of tests, not just the run-make ones :thinking:).

Reply:
Actually this is very good advice, I'll go back and revisit the example tests and the added wasm tests to see if I can make them better examples in terms of testing DSLs.

Reply:
auto-blessing is prone to being very dodgy to implement when the thing being tested is a bit more ad-hoc (ask me how I know lol)

Reply:
I don't think `.unstable_option(UnstableOpt::Pretty { kind: PrettyKind::HirTree })` is better than `-Zunpretty=hir-tree`, the latter is easier to understand *and* shorter. I agree for common flags like output file though, but would prefer not having a huge enum for every unstable opt 😅